tests/test_csc.py

```diff
@@ -0,0 +1,302 @@
+# This file is part of ts_FiberSpectrograph.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import unittest
+import shutil
+
+import asynctest
+
+from lsst.ts import salobj
+from lsst.ts.FiberSpectrograph import AvsSimulator
+from lsst.ts.FiberSpectrograph import FiberSpectrographCsc
+from lsst.ts.FiberSpectrograph import AvsReturnError, AvsReturnCode
+from lsst.ts.idl.enums.FiberSpectrograph import ExposureState
+
+STD_TIMEOUT = 2  # standard command timeout (sec)
+LONG_TIMEOUT = 20  # timeout for starting SAL components (sec)
+
+
+class Harness:
+    """An configurable async context manager for setting up and starting a CSC,
+    and any other pieces it needs to talk to."""
+    def __init__(self, initial_state, config_dir=None):
+        salobj.test_utils.set_random_lsst_dds_domain()
```

Comment:
You probably don't need call since you also call this function in `setUp` (which is, in my opinion, the better place for it).

Reply:
Good catch. I think I stole that from ATDome, so it probably has the same issue.

Reply:
Actually ATDome only calls it in the Harness, not in setUp. I should probably switch that.